pkg/services/dashboards/database/database.go

```diff
@@ -340,11 +340,11 @@ func getExistingDashboardByTitleAndFolder(sess *db.Session, dash *dashboards.Das
 	}
```

Comment:
We're going to need to think about this whole function, right now it assumes that title is unique within an org and that there will be either 0 or 1 result, after this change it would be possible for there to be many results.

Reply:
right -- the slug is no longer a pseudo uid -- it is just for the display

Reply:
It looks like the only place that uses the [get by title query](https://github.com/grafana/grafana/blob/main/pkg/services/dashboards/models.go#L257) is [file provisioning ](https://github.com/grafana/grafana/blob/main/pkg/services/provisioning/dashboards/file_reader.go#L260)

I'm not sure what the correct behavior should be if there are two files with the same name (title!) in file provisioning 🤔.

Reply:
right now, provisioning just breaks down when you try to provision two dashboards with the same title _without_ uids, it prints this: https://github.com/grafana/grafana/blob/7e5544ab21da88ec2bf987ee6583fcf9bbb944bf/pkg/services/provisioning/dashboards/validator.go#L108

however, it does work when you provision dashboards with the same title but with distinct UIDs. should we explicitly mention this as a limitation? ie. provisioning doesn't support folders/dashboards with same titles unless you provide UIDs - seems reasonable to me

Reply:
i take back that it works when you set UIDs. the dashboards were provisioned, but I still see the warnings in logs

Reply:
after a quick chat with @frossano-grafana, I think this is an OK behavior for now - we haven't removed the UI constraint either, so the only way to create a dashboard with the same title will be through the API